agent-core/src/main/java/com/newrelic/agent/android/instrumentation/okhttp2/OkHttp2Instrumentation.java

```diff
@@ -55,6 +60,17 @@ public static Call newCall(OkHttpClient client, Request request) {
         return new CallExtension(client, request, client.newCall(request), transactionState);
     }
 
+    private static void addHeadersAsCustomAttribute(TransactionState transactionState, Request request) {
+
+        Map<String, String> headers = new HashMap<>();
+        for (String s : HttpHeaders.getInstance().getHttpHeaders()) {
+            if (request.headers().get(s) != null) {
```

Comment:
If `getHttpHeaders()` was guaranteed to return a non-null collection with non-null elements, this nullability check would not be needed.

Reply:
request.headers().get(s) can returns null and if we dont have any headers to add as custom attribute in that case it returns null.